### PR TITLE
net: build fixes for -fno-common

### PR DESCRIPTION
**Pull request by gebart**
These changes are necessary to build the rpl_udp example with `-fno-common`

See #2346

### CODE REVIEW

File: sys/net/network_layer/sixlowpan/ip.c [hunk 1]
```diff
@@ -44,12 +44,12 @@ char addr_str[IPV6_MAX_ADDR_STR_LEN];
 #define LLHDR_IPV6HDR_LEN           (LL_HDR_LEN + IPV6_HDR_LEN)
 #define IPV6_NET_IF_ADDR_BUFFER_LEN (NET_IF_MAX * IPV6_NET_IF_ADDR_LIST_LEN)
 
-uint8_t ip_send_buffer[BUFFER_SIZE];
+static uint8_t ip_send_buffer[BUFFER_SIZE];
 uint8_t buffer[BUFFER_SIZE];
```
Comment (gebart) [inline sys/net/network_layer/sixlowpan/ip.c]: This buffer is not very well named, maybe `sixlowpan_buffer` is a more fitting name?
OT: This is one of the oldest lines in this file.

https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#naming

**Comment by gebart** [comment]
@authmillenon and @thomaseichinger may want to take a look as well.

**Comment by authmillenon** [comment]
ACK in general, but keep in mind, that most of these files will be deprecated once #1680 will get merged.

**Comment by gebart** [comment]
@authmillenon I don't mind if this gets deleted tomorrow, this is just a stop gap. 

It would be good if this can be merged while waiting for the new network stack just so that we avoid especially the aliasing of the at86rf2xx buffer with the 6lowpan buffer as well as letting RIOT build with `-fno-common` (which helps in avoiding human errors).

**Comment by authmillenon** [comment]
Just wanted to be sure you know that ;). Merge at will, when Travis is okay with it (just kicked it)

**Pull request merged**